samples/matter/common/src/task_executor.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "task_executor.h"
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+constexpr size_t kTaskQueueSize = CONFIG_APP_TASK_QUEUE_SIZE;
+
+K_MSGQ_DEFINE(sTaskQueue, sizeof(TaskExecutor::Task), kTaskQueueSize, alignof(TaskExecutor::Task));
+
+void TaskExecutor::PostTask(const Task& task)
+{
+	if (k_msgq_put(&sTaskQueue, &task, K_NO_WAIT) != 0) {
```

Comment:
Actually, copying bytes of std::function to the message queue can lead to double-free if std::function allocates some resources on heap. There are ways to hack-around it, but imho replacing std::function with something that never allocates memory dynamically would be better. Again, this is not something that must be done in this PR, but just saying that removing std::function from this picture is not only code size optimization :).

Reply:
I will create a task for that since I need to understand better how the LambdaBridge works. :)